src/broccoli/glimmer-build.js

```diff
@@ -0,0 +1,173 @@
+const notifyAddonIncluded = require('ember-build-utilities/lib/addons/notify-included');
+const shouldIncludeAddon = require('ember-build-utilities/lib/addons/should-include');
+const existsSync = require('exists-sync');
+const p = require('ember-cli-preprocess-registry/preprocessors');
+const BroccoliSource = require('broccoli-source');
+const Funnel = require('broccoli-funnel');
+const { WatchedDir } = BroccoliSource;
+const {
+  isType,
+  preprocessTemplates,
+  preprocessMinifyCss,
+  setupRegistry,
+  defaultRegistry
+} = p;
+
+
+const DEFAULT_TS_OPTIONS = {
+  tsconfig: {
+    compilerOptions: {
+      target: "ESNext",
+      module: "es2015",
+      inlineSourceMap: true,
+      inlineSources: true,
+      moduleResolution: "node"
+    },
+    exclude: [
+      'node_modules',
+      '**/*.d.ts'
+    ]
+  }
+};
+
+let DEFAULT_CONFIG = {
+  outputPaths: {
+    app: {
+      html: 'index.html',
+    }
+  },
+  addons: {},
+};
+
+class GlimmerBuild extends AbstractBuild {
+  constructor(defaults, options = {}) {
+    super(defaults);
+    this.options = options;
+    let missingProjectMessage = 'You must pass through the default arguments passed into your ember-cli-build.js file when constructing a new GlimmerApp';
+    if (arguments.length === 0) {
+      throw new Error(missingProjectMessage);
+    }
+
+    if (!defaults.project) {
+      throw new Error(missingProjectMessage);
+    }
+
+    this.env = process.env.EMBER_ENV || 'development';
+    this.project = defaults.project;
+    this.name = this.project.name();
+    this.targets = options.targets;
+    this.registry = options.registry || p.defaultRegistry(this);
+
+    setupRegistry(this);
+    this._notifyAddonIncluded();
+
+    // this.outputPaths = this.buildOutputPaths(options);
+    // this.detectInvalidBlueprint(options);
+  }
+
+  _initOptions() {
+    return
+  }
+
+  _notifyAddonIncluded() {
+    this.initializeAddons();
+    let { blacklist, whitelist } = this.options.addons;
+    notifyAddonIncluded(this.project, blacklist, whitelist, (addon) => {
+      addon.app = this;
+
+      if (shouldIncludeAddon(addon, blacklist, whitelist)) {
+        if (addon.included) {
+          addon.included(this);
+        }
+
+        return addon;
+      }
+    });
+  }
+
+  _sourceTree(options) {
+    let srcTree = options.trees && options.trees.src;
+    if (typeof srcTree === 'string') {
+      srcTree = new WatchedDir(resolveLocal(this.project.root, srcTree));
+    } else if (!srcTree) {
+      let srcPath = resolveLocal(this.project.root, 'src');
+      srcTree = existsSync(srcPath) ? new WatchedDir(srcPath) : null;
+    }
+
+    if (srcTree) {
+      return new Funnel(srcTree, {
+        destDir: 'src'
+      });
+    }
+  }
+
+  _buildTrees(options) {
+    let source = this._sourceTree(options);
+    let ts = new GlimmerTemplatePrecompiler(source, {
+      rootName: this.project.pkg.name
+    });
+    ts = this.typeScriptTree(ts);
```

Comment:
We need to:

* ensure typescript compilation gets the config tree
* ensure typescript compilation has access to node_modules